src/TLuaInterpreter.cpp

```diff
@@ -15592,3 +15597,227 @@ int TLuaInterpreter::disableClickthrough(lua_State* L)
+    QString text;
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "spellSuggestWord: bad argument #1 type (word as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    } else {
+        text = QString::fromUtf8(lua_tostring(L, 1));
+    }
+
+    bool useUserDictionary = false;
+    if (lua_gettop(L) > 1) {
+        if (!lua_isboolean(L, 2)) {
+            lua_pushfstring(L, "spellSuggestWord: bad argument #2 type (check profile dictionary as boolean is optional {use 'false' or omit to check against system dictionary}, got %s!)", luaL_typename(L, 2));
+            return lua_error(L);
+        } else {
+            useUserDictionary = lua_toboolean(L, 2);
+            if (useUserDictionary && !hasUserDictionary) {
+                lua_pushnil(L);
+                lua_pushstring(L, "no user dictionary enabled in the preferences for this profile");
+                return 2;
+            }
+        }
+    }
+
+    char **wordList;
+    size_t wordCount = 0;
+    Hunhandle* handle = nullptr;
+    QByteArray encodedText;
+    if (useUserDictionary) {
+        handle = host.mpConsole->getHunspellHandle_user();
+        encodedText = text.toUtf8();
+    } else {
+        handle = host.mpConsole->getHunspellHandle_system();
+        encodedText = host.mpConsole->getHunspellCodec_system()->fromUnicode(text);
+    }
+    // CHECKME: Is there any danger of contention here - do we need to get mudlet::mDictionaryReadWriteLock locked for reading if we are accessing the shared user dictionary?
+    wordCount = Hunspell_suggest(handle, &wordList, encodedText.constData());
+    lua_newtable(L);
+    for (size_t i = 0; i < wordCount; ++i) {
+        lua_pushnumber(L, i+1);
+        lua_pushstring(L, wordList[i]);
+        lua_settable(L, -3);
+    }
+    Hunspell_free_list(handle, &wordList, wordCount);
+    return 1;
+}
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Text_Functions#getDictionaryWordList
+int TLuaInterpreter::getDictionaryWordList(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+    bool hasUserDictionary = false;
+    bool hasSharedDictionary = false;
+    host.getUserDictionaryOptions(hasUserDictionary, hasSharedDictionary);
+    if (!host.mEnableSpellCheck) {
+        lua_pushnil(L);
+        lua_pushstring(L, "spell-checking is not enabled in the preferences for this profile");
+        return 2;
+    }
+
+    if (!hasUserDictionary) {
+        lua_pushnil(L);
+        lua_pushstring(L, "no user dictionary enabled in the preferences for this profile");
+        return 2;
+    }
+
+    // This may stall if this is accessing the shared user dictionary and that
+    // is being updated by another profile, but it should eventually return...
+    QStringList wordList = host.mpConsole->getWordSet().toList();
+    int wordCount = wordList.size();
+    if (wordCount > 1) {
+        QCollator sorter;
+        sorter.setCaseSensitivity(Qt::CaseSensitive);
```

Comment:
Why case-sensitive? Case-insensitive is more people-friendly

Reply:
:thinking: Actually, I am not sure - and the Qt documentation does not indicate which is the default...

... okay will change to `Qt::CaseInsensitve`.

Reply:
in next commit.